quarkus/service/src/main/java/org/apache/polaris/service/quarkus/auth/internal/InternalAuthenticationMechanism.java

```diff
@@ -56,9 +84,34 @@ public Uni<SecurityIdentity> authenticate(
     }
 
     String credential = authHeader.substring(spaceIdx + 1);
+
+    DecodedToken token;
+    try {
+      token = decodeToken(credential);
+    } catch (Exception e) {
+      return configuration.type() == AuthenticationType.MIXED
```

Comment:
Would it be more correct to only parse the token here and delegate validation to `InternalIdentityProvider`?

With that approach, I guess we will not have condition failure logic... All nuance should be handled by Quarkus, WDYT?

We could have a Polaris-specific JWT claim to identify Polaris tokens, I guess.

Reply:
Well, the issue is that "parse" and "validate" are basically tied together.

Auth0 has two methods: `JWT.decode` and `JWT.require`: the former decodes without validating the signature, the latter decodes and validates.

Currently, `JWTBroker.verify()` uses `JWT.require`. We _could_ introduce a new method in `TokenBroker`, e.g. `parse()` or `decode()`, and call `JWT.decode` there. But I don't know if it's worth the hassle: I bet that decoding the token is just slightly faster than decoding and verifying.

BTW that's why I introduced the `MIXED` authentication type: so that the extra penalty of decoding the token twice is only paid by realms that opt for that authentication type.

Reply:
After playing with this a bit, I'd be in favor of leaving this for a follow-up PR. The changes to the `TokenBroker` interface become a bit invasive imho.

I was trying something like this:

```java
interface TokenBroker {
  DecodedToken decode(String token); // done in InternalAuthenticationMechanism
  void verify(DecodedToken token); // done in InternalIdentityProvider
...
}
```

But:

1. `decode` would still throw an exception on every request with an external token;
2. We'd need to leak some auth0 types in the `DecodedToken` interface in order to avoid re-parsing the token. e.g. `DecodedJWT getAuth0Token();`

Reply:
Hmm actually my idea won't even work. We need to verify the signature as well in `InternalAuthenticationMechanism`. That's the _only_ way to be sure that the token is or isn't internal to Polaris.

Reply:
I was thinking that we could parse without validation in `InternalAuthenticationMechanism`, check a Polaris-specific property and trust it to choose the internal validation logic. Then, `InternalIdentityProvider` would finish the validation by checking the signature.

In any case, I think it makes sense to defer this.

Reply:
Let's do that separately.. This PR is already big and reviewed :)